src/main/java/Entities/Ingredient.java

```diff
@@ -18,7 +20,12 @@ public Ingredient(String name, List<Tag> tags) {
 
     @Override
     public Boolean has(Tag tag) {
-        return null;
+        for (Tag item : this.tags) {
+            if (tag.equals(item)) {
+                return true;
```

Comment:
We should override the Tag.equal method to ensure it works the way we want (can be done later).